lib/src/generators/client.dart

```diff
@@ -188,23 +189,27 @@ class $clientException implements io.HttpException {
 class $clientName {
   $clientName({
     $authInputCode
-    String? baseUrl,
+    this.baseUrl,
+    this.globalHeaders = const {},
```

Comment:
@davidmigloz can this just be called `headers` and have a doc string that states headers defined here will be global to all requests?

PS: Can you add a doc string to the class constructor for this as well? Enables someone to hover over the class name and get a quick summary of all the parameters and what they do:

```dart
/// Client for X
///
/// `baseUrl`: Override baseUrl URL - else defaults to server url defined in spec
///
/// `client`: Override HTTP client to use for requests
class XClient {
```

Reply:
You are right, the prefix `global` does not really add any value, renamed!

I've also improved the documentation.

Eg. of class documentation:
<img width="706" alt="image" src="https://github.com/tazatechnology/openapi_spec/assets/6546265/d1d9d530-d178-4a81-a46f-64592c3e7ab7">

Eg. of constructor documentation:
<img width="648" alt="image" src="https://github.com/tazatechnology/openapi_spec/assets/6546265/eef929c9-5987-4d53-b132-f11f001cebd0">

Reply:
nice!